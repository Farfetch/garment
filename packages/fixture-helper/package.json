{
  "name": "@garment/fixture-helper",
  "license": "MIT",
  "version": "0.16.0",
  "main": "lib/index.js",
  "files": [
    "lib"
  ],
  "dependencies": {
    "@garment/logger": "^0.16.0",
    "@garment/runner": "^0.16.0",
    "@garment/workspace": "^0.16.0",
    "fs-extra": "8.1.0",
    "globby": "10.0.1",
    "strip-ansi": "^5.2.0",
    "tempy": "0.3.0"
  },
  "devDependencies": {
    "is-valid-path": "^0.1.1",
    "normalize-path": "^3.0.0"
  }
}
