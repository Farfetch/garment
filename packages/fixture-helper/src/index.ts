export * from './executeRunner';
export * from './initFixtureHelper';
export * from './resolver';
import normalizePath = require('normalize-path');
import isValidPath = require('is-valid-path');

const containsValidPath = (value: string): boolean =>
  value.split(/\s/).some((s) => isValidPath(s));

export const normalizeContainedPath = (value: string): string => {
  return value
    .split(/(\s)/)
    .map((s) => {
      if (isValidPath(s)) {
        return normalizePath(s);
      } else {
        return s;
      }
    })
    .join('');
};

export function replaceTestPath<T>(
  obj: T,
  testPath: string,
  replaceWith = '/test_path'
): T {
  const normalizedTestPathRegEx = new RegExp(normalizePath(testPath), 'g');
  return JSON.parse(
    JSON.stringify(obj, (_, value) =>
      typeof value === 'string' && containsValidPath(value)
        ? normalizeContainedPath(value).replace(
            normalizedTestPathRegEx,
            replaceWith
          )
        : value
    )
  );
}
