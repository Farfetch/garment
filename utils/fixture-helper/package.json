{
  "name": "@garment/fixture-helper",
  "license": "MIT",
  "version": "0.14.0",
  "main": "lib/index.js",
  "files": [
    "lib"
  ],
  "dependencies": {
    "@garment/logger": "^0.14.0",
    "@garment/runner": "^0.14.0",
    "@garment/workspace": "^0.14.0",
    "fs-extra": "8.1.0",
    "globby": "10.0.1",
    "strip-ansi": "^5.2.0",
    "tempy": "0.3.0"
  },
  "devDependencies": {
    "is-valid-path": "^0.1.1",
    "normalize-path": "^3.0.0"
  }
}
