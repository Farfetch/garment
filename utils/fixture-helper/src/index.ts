export * from './executeRunner';
export * from './initFixtureHelper';

export function replaceTestPath<T>(
  obj: T,
  testPath: string,
  replaceWith = '/test_path'
): T {
  return JSON.parse(
    JSON.stringify(obj, (_, value) =>
      typeof value === 'string'
        ? value.replace(new RegExp(testPath, 'g'), replaceWith)
        : value
    )
  );
}
