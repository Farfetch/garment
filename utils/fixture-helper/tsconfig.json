{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "rootDir": "src",
    "outDir": "lib",
    "composite": true
  },
  "exclude": [
    "node_modules",
    "lib",
    "__tests__"
  ],
  "references": [
    {
      "path": "../../core/logger"
    },
    {
      "path": "../../core/runner"
    },
    {
      "path": "../../core/workspace"
    }
  ]
}