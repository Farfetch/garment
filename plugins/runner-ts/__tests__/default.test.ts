import { executeRunner2, initFixtureHelper } from '@garment/fixture-helper';
import * as Path from 'path';
import handler from '../src';

const { initFixture, clean } = initFixtureHelper(module, {
  tempDir: Path.join(__dirname, 'tmp__')
});

afterAll(clean);

test('Transpiles TypeScript', async () => {
  const cwd = await initFixture('basic');

  const src = (p = '') => Path.join(cwd, 'test-project', 'src', p);

  const result = await executeRunner2(
    handler,
    { configFile: 'tsconfig.json' },
    {
      projectRelativePath: 'test-project',
      cwd,
      files: file => [
        file.text('index.ts', 'export const foo: number = 420;', {
          absolutePath: src('nested/index.ts'),
          baseDir: src()
        })
      ]
    }
  );

  expect(result).toMatchSnapshot();
});

test('Transpiles TypeScript with relative imports including not provided as input', async () => {
  const cwd = await initFixture('basic');

  const src = (p = '') => Path.join(cwd, 'test-project', 'src', p);

  const result = await executeRunner2(
    handler,
    { configFile: 'tsconfig.json' },
    {
      projectRelativePath: 'test-project',
      cwd,
      files: file => [
        file.text('nested/index.ts', 'export const foo: number = 420;', {
          absolutePath: src('nested/index.ts'),
          baseDir: src()
        }),
        file.text(
          'main/index.ts',
          'import { foo } from "../nested"; console.log(foo);',
          { absolutePath: src('main/index.ts'), baseDir: src() }
        )
      ]
    }
  );

  expect(result).toMatchSnapshot();
});

test('Transpiles TypeScript', async () => {
  const cwd = await initFixture('config-extends');

  const src = (p = '') => Path.join(cwd, 'test-project', 'src', p);

  const result = await executeRunner2(
    handler,
    { configFile: 'tsconfig.json' },
    {
      projectRelativePath: 'test-project',
      cwd,
      files: file => [
        file.text(
          'index.ts',
          'export const greet = (str: string) => `Hello, ${str}`',
          {
            absolutePath: src('nested/index.ts'),
            baseDir: src()
          }
        )
      ]
    }
  );

  expect(result).toMatchSnapshot();
});

test('Transpiles TypeScript when configFile contains npm module in extends field', async () => {
  const cwd = await initFixture('config-extends-npm-module', { staticId: 0 });

  const src = (p = '') => Path.join(cwd, 'test-project', 'src', p);

  const result = await executeRunner2(
    handler,
    { configFile: 'tsconfig.json' },
    {
      projectRelativePath: 'test-project',
      cwd,
      files: file => [
        file.text(
          'index.ts',
          'export const greet = (str: string) => `Hello, ${str}`',
          {
            absolutePath: src('nested/index.ts'),
            baseDir: src()
          }
        )
      ]
    }
  );

  expect(result).toMatchSnapshot();
});
