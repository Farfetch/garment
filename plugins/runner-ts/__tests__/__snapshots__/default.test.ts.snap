// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transpiles TypeScript 1`] = `
Object {
  "collectedOutput": Array [
    Object {
      "cacheKeys": Array [
        "3.8.3",
        "{\\"target\\":\\"es5\\",\\"module\\":\\"commonjs\\",\\"moduleResolution\\":\\"node\\",\\"declaration\\":true,\\"sourceMap\\":true,\\"noEmitOnError\\":true,\\"skipLibCheck\\":true}",
        "export const foo: number = 420;",
      ],
      "dependencies": Array [],
      "files": Array [
        Object {
          "data": "{\\"version\\":3,\\"file\\":\\"index.js\\",\\"sourceRoot\\":\\"\\",\\"sources\\":[\\"index.ts\\"],\\"names\\":[],\\"mappings\\":\\";;AAAa,QAAA,GAAG,GAAW,GAAG,CAAC\\"}",
          "path": "nested/index.js.map",
          "type": "text",
        },
        Object {
          "data": "\\"use strict\\";
Object.defineProperty(exports, \\"__esModule\\", { value: true });
exports.foo = 420;
//# sourceMappingURL=index.js.map",
          "path": "nested/index.js",
          "type": "text",
        },
        Object {
          "data": "export declare const foo: number;
",
          "path": "nested/index.d.ts",
          "type": "text",
        },
      ],
      "logs": Array [],
    },
  ],
  "dependsOnFile": Array [
    "/test_path/tsconfig.json",
  ],
  "isLongRunning": false,
  "logs": Array [
    Object {
      "args": Array [
        "[TS] Processing",
        "index.ts",
      ],
      "level": "debug",
    },
  ],
  "onDestroyHandler": undefined,
  "shouldWatchDependencies": false,
}
`;

exports[`Transpiles TypeScript 2`] = `
Object {
  "collectedOutput": Array [
    Object {
      "cacheKeys": Array [
        "3.8.3",
        "{\\"declaration\\":true,\\"sourceMap\\":true,\\"noEmitOnError\\":true,\\"skipLibCheck\\":true,\\"target\\":\\"es5\\",\\"module\\":\\"esnext\\",\\"moduleResolution\\":\\"node\\"}",
        "export const greet = (str: string) => \`Hello, \${str}\`",
      ],
      "dependencies": Array [],
      "files": Array [
        Object {
          "data": "{\\"version\\":3,\\"file\\":\\"index.js\\",\\"sourceRoot\\":\\"\\",\\"sources\\":[\\"index.ts\\"],\\"names\\":[],\\"mappings\\":\\"AAAA,MAAM,CAAC,IAAM,KAAK,GAAG,UAAC,GAAW,IAAK,OAAA,YAAU,GAAK,EAAf,CAAe,CAAA\\"}",
          "path": "nested/index.js.map",
          "type": "text",
        },
        Object {
          "data": "export var greet = function (str) { return \\"Hello, \\" + str; };
//# sourceMappingURL=index.js.map",
          "path": "nested/index.js",
          "type": "text",
        },
        Object {
          "data": "export declare const greet: (str: string) => string;
",
          "path": "nested/index.d.ts",
          "type": "text",
        },
      ],
      "logs": Array [],
    },
  ],
  "dependsOnFile": Array [
    "/test_path/tsconfig.json",
    "/test_path/tsconfig.base.json",
    "/test_path/common/tsconfig.common.json",
  ],
  "isLongRunning": false,
  "logs": Array [
    Object {
      "args": Array [
        "[TS] Processing",
        "index.ts",
      ],
      "level": "debug",
    },
  ],
  "onDestroyHandler": undefined,
  "shouldWatchDependencies": false,
}
`;

exports[`Transpiles TypeScript when configFile contains npm module in extends field 1`] = `
Object {
  "collectedOutput": Array [
    Object {
      "cacheKeys": Array [
        "3.8.3",
        "{\\"declaration\\":true,\\"sourceMap\\":true,\\"noEmitOnError\\":true,\\"skipLibCheck\\":true,\\"target\\":\\"es5\\",\\"module\\":\\"esnext\\",\\"moduleResolution\\":\\"node\\"}",
        "export const greet = (str: string) => \`Hello, \${str}\`",
      ],
      "dependencies": Array [],
      "files": Array [
        Object {
          "data": "{\\"version\\":3,\\"file\\":\\"index.js\\",\\"sourceRoot\\":\\"\\",\\"sources\\":[\\"index.ts\\"],\\"names\\":[],\\"mappings\\":\\"AAAA,MAAM,CAAC,IAAM,KAAK,GAAG,UAAC,GAAW,IAAK,OAAA,YAAU,GAAK,EAAf,CAAe,CAAA\\"}",
          "path": "nested/index.js.map",
          "type": "text",
        },
        Object {
          "data": "export var greet = function (str) { return \\"Hello, \\" + str; };
//# sourceMappingURL=index.js.map",
          "path": "nested/index.js",
          "type": "text",
        },
        Object {
          "data": "export declare const greet: (str: string) => string;
",
          "path": "nested/index.d.ts",
          "type": "text",
        },
      ],
      "logs": Array [],
    },
  ],
  "dependsOnFile": Array [
    "/test_path/tsconfig.json",
    "/test_path/node_modules/@my-scope/tsconfig/tsconfig.base.json",
    "/test_path/node_modules/@my-scope/tsconfig/tsconfig.common.json",
  ],
  "isLongRunning": false,
  "logs": Array [
    Object {
      "args": Array [
        "[TS] Processing",
        "index.ts",
      ],
      "level": "debug",
    },
  ],
  "onDestroyHandler": undefined,
  "shouldWatchDependencies": false,
}
`;

exports[`Transpiles TypeScript with relative imports including not provided as input 1`] = `
Object {
  "collectedOutput": Array [
    Object {
      "cacheKeys": Array [
        "3.8.3",
        "{\\"target\\":\\"es5\\",\\"module\\":\\"commonjs\\",\\"moduleResolution\\":\\"node\\",\\"declaration\\":true,\\"sourceMap\\":true,\\"noEmitOnError\\":true,\\"skipLibCheck\\":true}",
        "export const foo: number = 420;",
      ],
      "dependencies": Array [],
      "files": Array [
        Object {
          "data": "{\\"version\\":3,\\"file\\":\\"index.js\\",\\"sourceRoot\\":\\"\\",\\"sources\\":[\\"index.ts\\"],\\"names\\":[],\\"mappings\\":\\";;AAAa,QAAA,GAAG,GAAW,GAAG,CAAC\\"}",
          "path": "nested/index.js.map",
          "type": "text",
        },
        Object {
          "data": "\\"use strict\\";
Object.defineProperty(exports, \\"__esModule\\", { value: true });
exports.foo = 420;
//# sourceMappingURL=index.js.map",
          "path": "nested/index.js",
          "type": "text",
        },
        Object {
          "data": "export declare const foo: number;
",
          "path": "nested/index.d.ts",
          "type": "text",
        },
      ],
      "logs": Array [],
    },
    Object {
      "cacheKeys": Array [
        "3.8.3",
        "{\\"target\\":\\"es5\\",\\"module\\":\\"commonjs\\",\\"moduleResolution\\":\\"node\\",\\"declaration\\":true,\\"sourceMap\\":true,\\"noEmitOnError\\":true,\\"skipLibCheck\\":true}",
        "import { foo } from \\"../nested\\"; console.log(foo);",
        "export const foo: number = 420;",
      ],
      "dependencies": Array [
        "/test_path/test-project/src/nested/index.ts",
      ],
      "files": Array [
        Object {
          "data": "{\\"version\\":3,\\"file\\":\\"index.js\\",\\"sourceRoot\\":\\"\\",\\"sources\\":[\\"index.ts\\"],\\"names\\":[],\\"mappings\\":\\";;AAAA,oCAAgC;AAAC,OAAO,CAAC,GAAG,CAAC,YAAG,CAAC,CAAC\\"}",
          "path": "main/index.js.map",
          "type": "text",
        },
        Object {
          "data": "\\"use strict\\";
Object.defineProperty(exports, \\"__esModule\\", { value: true });
var nested_1 = require(\\"../nested\\");
console.log(nested_1.foo);
//# sourceMappingURL=index.js.map",
          "path": "main/index.js",
          "type": "text",
        },
        Object {
          "data": "export {};
",
          "path": "main/index.d.ts",
          "type": "text",
        },
      ],
      "logs": Array [],
    },
  ],
  "dependsOnFile": Array [
    "/test_path/tsconfig.json",
  ],
  "isLongRunning": false,
  "logs": Array [
    Object {
      "args": Array [
        "[TS] Processing",
        "nested/index.ts",
      ],
      "level": "debug",
    },
    Object {
      "args": Array [
        "[TS] Processing",
        "main/index.ts",
      ],
      "level": "debug",
    },
  ],
  "onDestroyHandler": undefined,
  "shouldWatchDependencies": false,
}
`;
